tool_imageoptimize.php

```diff
@@ -349,23 +349,32 @@ public function handle(string $action) : bool {
                         if (file_exists($fromFilePath)) {
                             $optimizerChain = OptimizerChainFactory::create();
                             $optimizerChain->optimize($fromFilePath, $toFilePath);
-                            $toFileContent = file_get_contents($toFilePath);
-                            $this->filerecord->pathnamehash = \file_storage::get_pathname_hash(
-                                $this->filerecord->contextid,
-                                $this->filerecord->component,
-                                $this->filerecord->filearea,
-                                $this->filerecord->itemid,
-                                $this->filerecord->filepath,
-                                $this->filerecord->filename
-                            );
-                            list($this->filerecord->contenthash, $this->filerecord->filesize, $newFile) = $fileSystem
-                                ->add_file_from_string($toFileContent);
-                            if ($newFile) {
-                                $this->db->update_record('files', $this->filerecord);
-                                @unlink($fromFileSourcePath);
+
+                            if (file_exists($toFilePath) && filesize($toFilePath) > 0) {
+                                $originalhash = $this->filerecord->contenthash;
+                                $newhash = sha1_file($toFilePath);
+
+                                global $CFG;
+                                $newpath = $CFG->dataroot . '/filedir/' . substr($newhash,0,2) . '/' . substr($newhash,2,2) . '/';
```

Comment:
You should never interact with file api files directly, you cannot assume they are even on disk (ie they might be managed in s3 or somewhere else)

Reply:
thats a good point. In that case we would need to loop through all occurrences of this contenthash in the filestable, create a fileobject and update it using the file-api, wouldn't we?